Source/WebKit/UIProcess/WebProcessPool.cpp

```diff
@@ -1954,7 +1944,8 @@ std::tuple<Ref<WebProcessProxy>, SuspendedPageProxy*, ASCIILiteral> WebProcessPo
     if (m_automationSession)
         return { WTFMove(sourceProcess), nullptr, "An automation session is active"_s };
 
-    if (!sourceProcess->hasCommittedAnyProvisionalLoads()) {
+    // FIXME: We ought to be able to re-use processes that haven't committed anything with site isolation enabled, but cross-site redirects are tricky.
```

Comment:
I don't understand why this is an issue. How is it harder to use a process is fresh (has never done any loads) for a new navigation?

I suspect we're working around a pretty bad bug here.

Reply:
If we reuse a process that hasn't had any committed loads, then when an iframe redirects across sites we get navigationChangesFrameProcess being false when it should be true and we never call continueNavigationInNewProcess which never calls prepareForProvisionalNavigationInProcess and it doesn't continue with the redirect.  I'm sure there's a way to resolve this, but I think it can be done in a separate PR.